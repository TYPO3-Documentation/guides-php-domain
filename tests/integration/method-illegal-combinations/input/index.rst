=============================================
PHP method with allowed modifier combinations
=============================================

..  php:method:: doSomething1(string $whatever): string
    :abstract:
    :final:

    Do something

..  php:method:: doSomething2(string $whatever): string
    :public:
    :protected:

    Do something

..  php:method:: doSomething3(string $whatever): string
    :public:
    :private:

    Do something

..  php:method:: doSomething4(string $whatever): string
    :protected:
    :private:

    Do something

..  php:method:: doSomething5(string $whatever): string
    :private:
    :abstract:

    Do something
