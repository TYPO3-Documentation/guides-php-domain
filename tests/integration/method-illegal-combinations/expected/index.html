<!-- content start -->
    <div class="section" id="php-method-with-allowed-modifier-combinations">
    <h1>PHP method with allowed modifier combinations</h1>

            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething1"><span class="sig-name modifier"><span class="pre">abstract</span></span>
 <span class="sig-name modifier"><span class="pre">final</span></span>
 <span class="sig-name modifier"><span class="pre">final</span></span>

<span class="sig-name descname"><span class="pre">doSomething1</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
</dt>
    <dd>
        <p>Do something</p>
    </dd>
</dl>

            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething2"><span class="sig-name modifier"><span class="pre">public</span></span>
 <span class="sig-name modifier"><span class="pre">protected</span></span>

<span class="sig-name descname"><span class="pre">doSomething2</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
</dt>
    <dd>
        <p>Do something</p>
    </dd>
</dl>

            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething3"><span class="sig-name modifier"><span class="pre">public</span></span>
 <span class="sig-name modifier"><span class="pre">private</span></span>

<span class="sig-name descname"><span class="pre">doSomething3</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
</dt>
    <dd>
        <p>Do something</p>
    </dd>
</dl>

            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething4"><span class="sig-name modifier"><span class="pre">protected</span></span>
 <span class="sig-name modifier"><span class="pre">private</span></span>

<span class="sig-name descname"><span class="pre">doSomething4</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
</dt>
    <dd>
        <p>Do something</p>
    </dd>
</dl>

            <dl class="php method">
    <dt class="sig sig-object php" id="dosomething5"><span class="sig-name modifier"><span class="pre">abstract</span></span>
 <span class="sig-name modifier"><span class="pre">private</span></span>

<span class="sig-name descname"><span class="pre">doSomething5</span></span>
<span class="sig-paren">(</span>
<em class="sig-param"><span class="pre">string $whatever</span></em><span class="sig-paren">)</span>
<em class="sig-returns"><span class="pre">: string</span></em>
</dt>
    <dd>
        <p>Do something</p>
    </dd>
</dl>

    </div>

<!-- content end -->
